code.py

```diff
@@ -17,9 +17,23 @@ def __init__(self, numBots, lines, pose):
 		for index, val in enumerate(self.lines):
 			self.group[index % self.bots] = np.append(self.group[index % self.bots], np.array([val]), axis = 0)
 		print(self.group)
-
+		
+		#length of the lines
+		self.length = []
+		for l in lines:
+			x1 = l[0][0]
+			y1 = l[0][1]
+			x2 = l[1][0]
+			y2 = l[1][1]
+			
+			#print (x1,y1,x2,y2)
+			dist = math.sqrt((x2 - x1)**2 + (y2 - y1)**2)
+			self.length.append(dist)
+			
 	# Initialize population
 	def initPop(self):
+		self.population = np.random.random_integers(low = 0, high = self.bots-1, size = (self.bots, self.numLines))
```

Comment:
Shouldn't size be (self.popSize, self.numLines)?

Reply:
Here we are defining the population right, so its basically the number of possible solutions we want. self.popSize is not being defined right, we are defining the population here. So what should I put it in its place.

Reply:
Your population consists of 'm' individuals/solutions, out of which you find the best one. Check the paper again.

Reply:
Yeah, although it is something we define by ourselves and not as such dependent on any data, so what should i define it as?